iptables/table.go

```diff
@@ -605,6 +622,7 @@ func (t *Table) loadDataplaneState() {
 
 	t.logCxt.Debug("Finished loading iptables state")
 	t.chainToDataplaneHashes = dataplaneHashes
+	t.chainToFullRules = dataplaneRules
```

Comment:
Think there's a problem just above here now that we're filtering out non-top-level chains.  The logic around line 604 (https://github.com/projectcalico/felix/blob/a11bdf65f2e44d202c1459aba0ac1df03e44df53/iptables/table.go#L604 scans _all_ chains for calico insertions and it may add chains to `t.dirtyInserts` that we haven't loaded the rules for. 

Wonder if we need to take another approach.  Maybe refactor `readHashesAndRulesFrom` so that it:

- always build up a `fullRules` slice (for non-calico chains)
- checks non-calico chains for inserts as it goes
- if it finds an insert, stores the `fullRules` slice in `rules`
- otherwise, throws `fullRules` away.

Reply:
I've fixed that as suggested but we'll be slurping all the KUBE* rules. Do you think it's safe to filter out the known kube-proxy chains?

Reply:
Yes, that's a bit frustrating but avoiding it completely would need a big refactoring, I think.  

We could do what we for hashes: store `""` for any non-Calico rules.  At least that'd mean that we wouldn't store a copy of all of kube-proxy's rules.